src/main/scala/is/hail/asm4s/Code.scala

```diff
@@ -106,13 +106,13 @@ object Code {
       def emit(il: Growable[AbstractInsnNode]): Unit = {
         val l1 = new LabelNode
         val l2 = new LabelNode
+        val l3 = new LabelNode
         il += l1
-        condition.emit(il)
-        il += new LdcInsnNode(0)
-        il += new JumpInsnNode(IF_ICMPEQ, l2)
+        condition.toConditional.emitConditional(il, l2, l3)
+        il += l2
         body.emit(il)
         il += new JumpInsnNode(GOTO, l1)
-        il += l2
+        il += l3
```

Comment:
Nice!

In the future, it'd be great if little things like this were their own PRs. That way we reduce the time until an improvement is in the mainline and also reduce overall work of the reviewer.